examples/dtls-sock/dtls-client.c

```diff
@@ -115,34 +171,48 @@ static int client_send(char *addr_str, char *data, size_t datalen)
         remote.netif = netif_get_id(netif);
     }
 
-    res = credman_add(&credential0);
-    if (res < 0 && res != CREDMAN_EXIST) {
-        /* ignore duplicate credentials */
-        printf("Error cannot add credential to system: %" PRIdSIZE "\n", res);
-        sock_udp_close(&udp_sock);
+    if (sock_udp_create(&udp_sock, &local, NULL, 0) < 0) {
+        puts("Error creating UDP sock");
         return -1;
     }
 
-#if IS_ACTIVE(CONFIG_DTLS_PSK)
-    /* register a second PSK credential */
-    res = credman_add(&credential1);
-    if (res < 0 && res != CREDMAN_EXIST) {
-        /* ignore duplicate credentials */
-        printf("Error cannot add credential to system: %" PRIdSIZE "\n", res);
+    /*
+     * Currently DTLS sock needs one and only one credential for the
+     * initialization. Subsequent credentials are made available to the sock
+     * by means of `sock_dtls_add_credential`.
+     */
+    if (sock_dtls_create(&dtls_sock, &udp_sock, SOCK_DTLS_CLIENT_TAG_0,
+                         SOCK_DTLS_1_2, SOCK_DTLS_CLIENT) < 0) {
+        puts("Error creating DTLS sock");
         sock_udp_close(&udp_sock);
         return -1;
     }
-    tag = SOCK_DTLS_CLIENT_TAG_1;
-#endif
 
-    res = sock_dtls_establish_session(&udp_sock, &dtls_sock, &session, tag,
-                                      &local, &remote, buf, sizeof(buf));
```

Comment:
Why move away from the `sock_dtls_establish_session()` helper and open-code the handshake again?

Reply:
Because it doesn't allow me to register the extra credentials before initiating the handshake.

Reply:
Same with the callbacks

Reply:
tbh I didn't quite understand the requirement when adding that helper, so if you think the API can be improved that would be great - leaving the handshake to the user seemed like a bad example.

Reply:
Hm I don't feel like this example is bad. It's not overly complicated or lengthy, the reader should get along well.
While a single helper function is nice and clean, it is clear the existing one is insufficient for this particular task. Adjusting that helper would mess with an API that is out of scope for this PR.

* Can we resolve this comment on these terms? 
* Do you think the API needs a rework? If so, we should open an issue for that.
* Makes me wonder, should we integrate credman closer with dtls?

Reply:
Sure, this shouldn't block this PR, I just took the opportunity to try to get some open questions about credman / DTLS answered :sweat_smile: 

I'd very much welcome an overhaul of credman as it seems still a bit unclear what it actually is and what's it's purpose.

Maybe it would also fit into PSA crypto?